[MM-23530] Add deployer config file documentation (#184)
mattermost/mattermost-load-test-ng: docs/deployer_config.md

# Deployer Configuration

## ClusterName

*string*

> Name of the cluster. This will be prefixed to all instances in AWS that built with the configuration.

Reviewer: s/instances/resources

that _are_ built

## AppInstanceCount

> *integer*

Reviewer 2: ```suggestion
*int*
```

The number of Mattermost application instances.

## AgentCount

> *integer*

Reviewer 2: ```suggestion
*int*
```

> The number of load-test agent instances. The first agent instance will also run the coordinator.

Reviewer 2: ```suggestion
The number of load-test agent instances. The first instance will also host the [coordinator](coordinator.md).
```

## SSHPublicKey

*string*

> The path the SSH public key, this key is used for establishing an SSH connection to the AWS instances.

Reviewer: The path to the

## DBInstanceCount

> *integer*

Reviewer 2: ```suggestion
*int*
```

The number of Database instances.

## DBInstanceClass

*string*

Type of the Database instance. See types [here](https://aws.amazon.com/rds/instance-types/).

## DBInstanceEngine

*string*

> Type of the database backend. This can be either Postgres or MySQL.

Reviewer: We can just mention the values here - aurora-mysql or aurora-postgresql

## DBUserName

*string*

The username to connect to the database.

## DBPassword

*string*

The password to connect to the database.

## MattermostDownloadURL

*string*

The URL from where to download Mattermost release. This can also point to a local binary path if the user wants to run a load-test on a custom build. The path should be prefixed with `file://`. In that case, only the binary gets replaced, and the rest of the build comes from the latest stable release.

## MattermostLicenseFile

*string*

The location of the Mattermost Enterprise Edition license file.

## AdminEmail

*string*

> The e-mail will be used when creating a sysadmin user at the deployment process.

Reviewer 2: ```suggestion
The e-mail that will be used when creating a sysadmin user during the deployment process.
```

## AdminUsername

*string*

> The user name that will be used when creating a sysadmin user at the deployment process.

Reviewer 2: ```suggestion
The user name that will be used when creating a sysadmin user during the deployment process.
```

## AdminPassword

*string*

> The password that will be used when creating a sysadmin user at the deployment process.

Reviewer 2: ```suggestion
The password that will be used when creating a sysadmin user during the deployment process.
```

## GoVersion

*string*

Go version to download for compiling load-test source.

## SourceCodeRef

*string*

load-test-ng head reference.

## LogSettings

### EnableConsole

> *boolean*

Reviewer 2: ```suggestion
*bool*
```

> The load-test-ng head reference.

Reviewer 2: Can we expand a bit on this? How is this used? Why is it needed?

If true, the server outputs log messages to the console based on the ConsoleLevel option.

### ConsoleLevel

*string*

Level of detail at which log events are written to the console.

### ConsoleJson

> *boolean*

Reviewer 2: ```suggestion
*bool*
```

When true, logged events are written in a machine-readable JSON format. Otherwise, they are printed as plain text.

### EnableFile

> *boolean*

Reviewer 2: ```suggestion
*bool*
```

When true, logged events are written to the file specified by the `FileLocation` setting.

### FileLevel

*string*

Level of detail at which log events are written to log files.

### FileJson

> *boolean*

Reviewer 2: ```suggestion
*bool*
```
